docker-bake.hcl

```diff
@@ -0,0 +1,108 @@
+variable "DOCKER_REGISTRY" {default = "ghcr.io/openclarity"}
+variable "DOCKER_TAG" {default = "latest"}
+variable "SUFFIX" {default = ""}
+
+variable "VERSION" {default = null}
+variable "BUILD_TIMESTAMP" {default = null}
+variable "COMMIT_HASH" {default = null}
+variable "VMCLARITY_TOOLS_BASE" {default = null}
+
+function "get_tag" {
+  params = [name]
+  result = ["${DOCKER_REGISTRY}/${name}${SUFFIX}:${DOCKER_TAG}"]
+}
+
+group "default" {
+	targets = ["vmclarity-apiserver", "vmclarity-cli", "vmclarity-cr-discovery-server", "vmclarity-orchestrator", "vmclarity-ui", "vmclarity-ui-backend"]
+}
+
+group "e2e" {
+	targets = ["vmclarity-apiserver", "vmclarity-cli", "vmclarity-orchestrator", "vmclarity-ui", "vmclarity-ui-backend"]
+}
+
+target "_common" {
+	labels = {
+		"org.opencontainers.image.url" = "https://github.com/openclarity/vmclarity"
+		"org.opencontainers.image.licenses" = "Apache-2.0"
+	}
+	cache-to = ["type=local,dest=/tmp/.buildx-cache"]
+	cache-from = ["type=local,src=/tmp/.buildx-cache"]
```

Comment:
By default, the build is locally cached. It can be improved a bit with this, but I haven't noticed many changes. For CICD, this is definitely useful (GHA cache driver is supported but in beta, which can avoid the local dir cache driver). Not sure if you managed to benchmark this locally. 

I would prefer if these configurations were empty for local build, but configured for CICD only. WDYT?

Also, to enable local cache, I had to create another builder via:
```
$ docker buildx bake vmclarity-apiserver
# ERROR: Cache export is not supported for the docker driver.

$ docker buildx create --name container --driver=docker-container

$ docker buildx bake vmclarity-apiserver --builder=container
# ... works normally
```

Reply:
On second thought, @chrisgacsal had bad experiences with `gha` cache type, we should stick to `local`.